### ISSUE TITLE
"Limit Maximum Length" doesn't work

### ISSUE THREAD

**Issue by Idran** [doxxx/ffxiv-craft-opt-web#373]
The "Limit Maximum Length" setting doesn't seem to actually work as it should. Even when starting from an entirely clear simulation, the simulator will keep within the length limitation as it should at first. But as the processing continues, the sequence will slowly expand beyond the upper bound of that setting.

Example:

![image](https://user-images.githubusercontent.com/7596015/74098586-aedb6e00-4adf-11ea-9368-1b331fc1395b.png)

![image](https://user-images.githubusercontent.com/7596015/74098673-76885f80-4ae0-11ea-80cb-4c8dd113558f.png)

**Comment by Kirtanei** [comment]
If you increase the Penalty Weight to much larger values it will generally enforce the Maximum Length.

**Comment by Idran** [comment]
So it's not actually a strict minimum rejecting sequences above that length? The help text should probably be revised to make that explicit, then; from the name and description I assumed it would just reject long ones, not provide a strong bias towards ones below the limit.

**Comment by doxxx** [comment]
@Idran: Correct. It's actually very difficult to put a hard limit on it, especially when the solver is unable to find a solution that fits within the limit. I will update the help text as you suggest.

**Comment by doxxx** [comment]
I've updated the English help text (ID `LIMIT_MAX_LENGTH_INFO`) to better describe what it does: 

> Adds a penalty to sequences over the specified limit. This is not a strict limit; under certain conditions the solver may still generate a sequence that exceeds the limit.

Please could translators provide updated text for their languages? You can paste it here or in a PR. @AhmadAymanA99 @kurax @Jengines @butterflo @RyuaNerin. Thanks!

**Comment by doxxx** [comment]
I've deployed the updated English help text to the beta site.

**Comment by butterflo** [comment]
> @Idran: Correct. It's actually very difficult to put a hard limit on it, especially when the solver is unable to find a solution that fits within the limit. I will update the help text as you suggest.

https://github.com/doxxx/ffxiv-craft-opt-web/blob/master/app/js/ffxivcraftmodel.js#L1287

Just a question, how was the factor 0.1 determined here? For RHS `1.0 * maxActionsExceeded` or even `0.1 * maxActionsExceeded^2` might work better for enforcing constraints.

**Comment by doxxx** [comment]
@butterflo: Trial and error. The goal of the solver is to find a sequence that _completes_, so a higher penalty on the length that prevents it from finding a sequence that completes is undesirable. This factor seemed to work best.

Why do you need a strict maximum length limit?

**Comment by Jengines** [comment]
Hello, here is the French translation for this key : 


> Ajoute une pénalité aux séquences dépassant la limite spécifiée. Sous certaines conditions, le solveur peut toujours générer une séquence qui dépasse la limite.

If some other French transaltors can confirm it (or not).

**Comment by butterflo** [comment]
@doxxx:
> Trial and error. The goal of the solver is to find a sequence that _completes_, so a higher penalty on the length that prevents it from finding a sequence that completes is undesirable. This factor seemed to work best.
> 
> Why do you need a strict maximum length limit?

Often times I would want a macro sequence that fits within 14 lines, just so that I don't have to run a second macro. Not a huge issue though.
